.tekton/task.yaml

```
@@ -0,0 +1,167 @@
+spec:
+  params:
+  - name: revision
+    type: string
+  workspaces:
+    - name: task-pvc
+      mountPath: /workspace
+  steps:
+    - name: clone
+      image: alpine/git
+      script: |
+        #!/bin/sh
+        echo "Cloning repo"
+        cd /workspace && git clone --depth 1 -b $(params.revision) https://github.com/instana/python-sensor
+        ls -lah /workspace
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: python-tracer-unittest-cassandra-task
+spec:
+  sidecars:
+    - name: cassandra
+      image: cassandra:3.11
+      env:
+        - name: MAX_HEAP_SIZE
+          value: 2048m
+        - name: HEAP_NEWSIZE
+          value: 512m
+      readinessProbe:
+        exec:
+          command:
+          - cqlsh
+          - -e
+          - 'describe cluster'
+  params:
+  - name: imageTag
+    type: string
+  workspaces:
+    - name: task-pvc
+      mountPath: /workspace
+  steps:
+    - name: unittest
+      image: python:$(params.imageTag)
+      env:
+        - name: TEST_CONFIGURATION
+          value: cassandra
+      workingDir: /workspace/python-sensor/
+      command:
+      - /workspace/python-sensor/.tekton/run_unittests.sh
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: python-tracer-unittest-couchbase-task
+spec:
+  sidecars:
+    - name: couchbase
+      image: couchbase/server-sandbox:5.5.0
+      readinessProbe:
+        httpGet:
+          path: /ui/index.html
+          port: 8091
+        # This Couchbase image recommends 60sec waiting for initial configuration
+        # Starting the tests too soon may result in
+        # "Error during initial configuration - aborting container"
+        # apparently because "vbucket map not available yet"
+        initialDelaySeconds: 60
+  params:
+  - name: imageTag
+    type: string
+  workspaces:
+    - name: task-pvc
+      mountPath: /workspace
+  steps:
+    - name: unittest
+      image: python:$(params.imageTag)
+      env:
+        - name: TEST_CONFIGURATION
+          value: couchbase
+      workingDir: /workspace/python-sensor/
+      command:
+      - /workspace/python-sensor/.tekton/run_unittests.sh
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: python-tracer-unittest-gevent-task
+spec:
+  params:
+  - name: imageTag
+    type: string
+  workspaces:
+    - name: task-pvc
+      mountPath: /workspace
+  steps:
+    - name: unittest
+      image: python:$(params.imageTag)
+      env:
+        - name: TEST_CONFIGURATION
+          value: gevent
+      workingDir: /workspace/python-sensor/
+      command:
+      - /workspace/python-sensor/.tekton/run_unittests.sh
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: python-tracer-unittest-default-task
+spec:
+  sidecars:
+    - name: google-cloud-pubsub
+      image: egymgmbh/pubsub-emulator
+      command:
+        - /init.sh
+        - test-project
+        - test-topic
+        - test-subscription
+    - name: mariadb
+      image: mariadb:11.2.3
```

Why not remove the version (container tag) and use the most updated version here? With that, we can always guarantee to trace against the most updated server versions. 

```suggestion
      image: mariadb
```

---

For example the new pulls now have already exhausted some PQS limits when pulling from dockerhub:
`The sidecar "mongo" in TaskRun "python-tracer-pr-tekton-test-c2d58b8-unittest-default-1" failed to pull the image "". The pod errored with the message: "Back-off pulling image "mongo"."`
Among the pod events one can see:
````
  Normal  Killing    66s    kubelet            Container sidecar-mongo definition changed, will be restarted
  Normal  Killing    66s    kubelet            Container sidecar-postgres definition changed, will be restarted
````

---

I will experiment with a pull through proxy for dockerhub, that should help.

---

I understand Tekton's recommendation, and, IMHO, I agree with that in parts. 

I guess we can then use the digest of the latest images that work for us, keep tracking for the changes, and create PRs to update the digests if necessary.

---

All right then I am changing to digest here.